src/main/java/org/opentripplanner/routing/core/State.java

```diff
@@ -159,6 +161,14 @@ protected State clone() {
         State ret;
         try {
             ret = (State) super.clone();
+            // Needed a deep copy of the maps below to properly pass information about prices
```

Comment:
Z tego co wiem, to tutaj wystarczy zwykła kopia mapy, bo zarówno `Integer` jak i `BigDecimal` są niemutowalne. Dość kłopotliwe jest natomiast to, że w javie kopiowanie map było dość drogie, a my ją będziemy kopiowali przy każdym przejściu po krawędzi. Może lepiej jakby to była zwykła stara tablica? w końcu te indeksy paczek i tak chyba nigdy nie przekroczą 4...